artemis-commons/src/main/java/org/apache/activemq/artemis/utils/DefaultSensitiveStringCodec.java

```diff
@@ -79,6 +79,21 @@ public void init(Map<String, String> params) throws Exception {
       }
    }
 
+   @Override
+   public boolean compare(Object value, Object encodedValue) {
+      if (encodedValue instanceof String == false) {
+         throw new IllegalArgumentException("Not supported encodedValue type: " + encodedValue.getClass().getName());
+      }
```

Comment:
> doing '== boolean' in an if is pretty ugly. Seems like either using !(encodedValue instanceof String), or changing the if to wrap the overall comparison and throw at the end would be nicer.
Nice suggestion moving it at the end

> This could also NPE generating the exception, on encodedValue.getClass().
I would not add another condition to check if value is null just to throw NPE

> "Unsupported" would seem a better fit rather than "Not supported".
I'll replace Not supported with Unsupported